pkg/deployment/depresource/shutdown.go

```diff
@@ -0,0 +1,101 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package depresource
+
+import (
+	"errors"
+
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/cloud-sdk-go/pkg/client/deployments"
+	"github.com/hashicorp/go-multierror"
+
+	"github.com/elastic/ecctl/pkg/deployment/deputil"
+	"github.com/elastic/ecctl/pkg/util"
+)
+
+// ShutdownParams is consumed by Shutdown.
+type ShutdownParams struct {
+	*api.API
+
+	DeploymentID string
```

Comment:
I'm seeing that most of the resources commands have those four fields the same  (`API`,`DeploymentID`,`RefID`, `Type` ) and we perform the same validations 

Would it make sense to extract those for into a `CommonXTZParams` struct with the its own validation and embed it to other structs?

This will save us a lot of code and increase code re-use and avoidance of future errors ( for instance add one more check or field )